src/view/xwayland.cpp

```diff
@@ -327,11 +343,29 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
             update_decorated();
         });
 
+        on_set_hints.set_callback([&] (void*) {
+            /*
+             * By storing urgency data instead of a signal
+             * it can be manually "acknowledged" by removing the data
+            */
+            if (xw->hints_urgency && !this->has_data("view-demands-attention"))
+                this->store_data(std::unique_ptr<wf::custom_data_t>(), "view-demands-attention");            
```

Comment:
why is this useful? I think that emitting the signal is enough. Plugins can store this data themselves if they want/need to.

Reply:
kinda unsure since hints_change may represent !xw->hints_urgency at some point, than what? view-no-longer-demands-attention?

Reply:
put a flag in the signal?

Reply:
you're right, my usual thinking is more toward the goal, whether my mind is currently located in plugin or not. The (theoretical, sure let's not get ahead of ourselves) issue I have is this though,
this requires all plugins that use this to use the same semantics for the data i.e the data name,
which hasn't been 'defined' anywhere. I understand this may not seem real at this point but the virtual barrier this detail puts, well I can't unsee it...

Reply:
Nevertheless of course I will change it too a flag in the signal

Reply:
But each plugin that cares about this data will receive the signal, right? So all of them will know that the view has or doesn't have the urgent hint.